rst2html5: init at 1.9.3

AndersonTorres opened the pull request:
###### Motivation for this change

Adds rst2html5
Converts ReSTructuredtext to (X)HTML5

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

AndersonTorres committed:
rst2html5: init at 1.9.3

Converts ReSTructuredtext (.rst) to (X)HTML5

AndersonTorres commented:
There is something wrong with Travis?

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -22252,6 +22252,27 @@ in {
+    version = "1.8.2";
+
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/bc/33/86da22a1b2fc63403f19567d9e07be311881abeb3e0380d7f6d7faf8b8b6/${name}.tar.gz";
```
please don't use an url like this because it has to be changed each time when the version changes.

AndersonTorres replied:
I don't know another URL to that. Maybe something "automatically generated"?

FRidh replied:
One that always works is 
```
url = "mirror://pypi/${builtins.substring 0 1 pname}/${pname}/${name}.tar.gz";
```
where `pname` is the name of the package on PyPI, which should be the same as the name of the package, so, `rst2html5`
```
pname = "rst2html5";
version = "1.8.2";
name = "${pname}-${version}";

AndersonTorres replied:
I'll try now!

AndersonTorres commented:
ping @FRidh - there is something more to be made?

7c6f434c commented:
There always is; but the build failure seems genuine: you have the file already, so the fetch failure cannot be seen locally without removing it first. Travis hits a 404.

AndersonTorres commented:
Many thanks, @7c6f434c ! I have seen the error now (a silly typo), I will fix it and upload a new version.

7c6f434c commented:
`nix-store --delete "$(nix-build '<nixpkgs>' -A something.src --no-out-link)"`

7c6f434c commented:
And now I guess it is time for @FRidh to come and ask to move the code to a separate file, like it is done (now) with the new Python packages…

AndersonTorres committed:
rst2html5: put its expression in a separate file

As recommended by @7c6f434c and @FRidh, rst2html5 now resides
in its proper file.

AndersonTorres commented:
Done!
